src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -243,23 +357,40 @@ public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
         {
             try
             {
-                string? packageNamespace = purl?.Namespace?.Replace('.', '/');
                 string? packageName = purl?.Name;
-                string feedUrl = (purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT).EnsureTrailingSlash();
+                string? packageNamespace = purl?.Namespace?.Replace('.', '/');
+                string feedUrl = purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT.GetRepositoryUrl();
+                MavenSupportedUpstream upstream = feedUrl.GetMavenSupportedUpstream();
+
                 HttpClient httpClient = CreateHttpClient();
+
+                string version;
                 if (purl?.Version == null)
                 {
-                    foreach (string? version in await EnumerateVersionsAsync(purl, useCache))
+                    // if no version is specified, use the earliest version available
+                    var versions = await EnumerateVersionsAsync(purl, useCache);
+                    if (!versions.Any())
                     {
-                        return await GetHttpStringCache(httpClient, $"{feedUrl}{packageNamespace}/{packageName}/{version}/{packageName}-{version}.pom", useCache);
+                        throw new Exception("No version specified and unable to enumerate.");
                     }
-                    throw new Exception("No version specified and unable to enumerate.");
+
+                    version = versions.ElementAt(0);
                 }
                 else
                 {
-                    string version = purl.Version;
-                    return await GetHttpStringCache(httpClient, $"{feedUrl}{packageNamespace}/{packageName}/{version}/{packageName}-{version}.pom", useCache);
+                    version = purl.Version;
+                }
+
+                if (upstream == MavenSupportedUpstream.MavenCentralRepository)
+                {
+                    return await GetHttpStringCache(httpClient, $"{upstream.GetRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}/{version}/{packageName}-{version}.pom", useCache);
+                }
+                else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+                {
+                    return await GetHttpStringCache(httpClient, $"{upstream.GetDownloadRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}/{version}/{packageName}-{version}.pom", useCache);
                 }
```

Comment:
From my other comments on the extension methods, I think this could be consolidated into a single implementation.